galasa-managers-parent/galasa-managers-workflow-parent/dev.galasa.githubissue.manager/src/main/java/dev/galasa/githubissue/internal/GitHubIssueManagerImpl.java

```diff
@@ -84,21 +89,23 @@ public void initialise(@NotNull IFramework framework, @NotNull List<IManager> al
 		if (galasaTest.isJava()) {
 			Class<?> testClass = galasaTest.getJavaTestClass();
 			
-			this.gitHubIssue = testClass.getAnnotation(GitHubIssue.class); // Set globally as might need later
-			if (this.gitHubIssue == null) { 
-				
-				GitHubIssue gitHubIssue = null;
+			this.classGitHubIssue = testClass.getAnnotation(GitHubIssue.class); // Set globally as might need later
+			if (this.classGitHubIssue == null) { 
+				GitHubIssue methodGitHubIssue = null;
 				for (Method testMethod : testClass.getMethods()) {
-					gitHubIssue = testMethod.getAnnotation(GitHubIssue.class);
-					if (gitHubIssue != null) {
-						required = true;
+					methodGitHubIssue = testMethod.getAnnotation(GitHubIssue.class);
+					if (methodGitHubIssue != null) {
+						methodAnnotated = true;
 						break;
 					}
 				}
 				// Annotation not present on the class or any methods
-				if (!required) {
+				if (!methodAnnotated) {
 					return;
 				}
+			} 
+			if (testClass.getAnnotation(ContinueOnTestFailure.class) != null) {
+				this.continueOnTestFailure = true;
```

Comment:
we shouldn't read another classes annotations.  If the manager that provides this annotation changes it's behaviour (uses a CPS property or extra attribute) to determine its state, then that logic would need to be added here as well.  We need to provide an SPI method in the manager to allow this manager to ask if we are continuing on test failure or not.  Don't worry - I've made this mistake before, it's the easier way out but can you change it please ;)

Reply:
As above